ui/page/staking/overview.go

```diff
@@ -446,6 +489,124 @@ func (pg *Page) HandleUserInteractions() {
+		}
+	}
+
+	if pg.autoPurchaseSettings.Button.Clicked() {
+		if pg.WL.MultiWallet.IsAutoTicketsPurchaseActive(walID) {
+			return
+		}
+
+		pg.ticketBuyerSettingsModal()
+	}
+}
+
+func (pg *Page) ticketBuyerSettingsModal() {
+	newTicketBuyerModal(pg.Load).
+		SettingsSaved(func() {
+			pg.Toast.Notify("Auto ticket purchase setting saved successfully")
+		}).
+		CancelSave(func() {
+			pg.autoPurchase.SetChecked(false)
+		}).
+		Show()
+}
+
+func (pg *Page) startTicketBuyerPasswordModal() {
+	host, walID, acctNum, b2m := pg.WL.MultiWallet.GetAutoTicketsBuyerConfig()
+	balToMaintain := dcrlibwallet.AmountCoin(b2m)
+
+	modal.NewPasswordModal(pg.Load).
+		Title("Confirm Automatic Ticket Purchase").
+		SetCancelable(false).
+		ExtraLayout(func(gtx C) D {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(pg.Theme.Label(values.TextSize14, fmt.Sprintf("Balance to maintain: %2.f", balToMaintain)).Layout),
+				layout.Rigid(func(gtx C) D {
+					label := pg.Theme.Label(values.TextSize14, fmt.Sprintf("VSP: %s", host))
+					return layout.Inset{Bottom: values.MarginPadding12}.Layout(gtx, label.Layout)
+				}),
+				layout.Rigid(func(gtx C) D {
+					return decredmaterial.LinearLayout{
+						Width:      decredmaterial.MatchParent,
+						Height:     decredmaterial.WrapContent,
+						Background: pg.Theme.Color.LightBlue,
+						Padding: layout.Inset{
+							Top:    values.MarginPadding12,
+							Bottom: values.MarginPadding12,
+						},
+						Border:    decredmaterial.Border{Radius: decredmaterial.Radius(8)},
+						Direction: layout.Center,
+						Alignment: layout.Middle,
+					}.Layout2(gtx, func(gtx C) D {
+						return layout.Inset{Bottom: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
+							msg := "Godcr must remain running, for tickets to be automatically purchased"
+							txt := pg.Theme.Label(values.TextSize14, msg)
+							txt.Alignment = text.Middle
+							txt.Color = pg.Theme.Color.GrayText3
+							return txt.Layout(gtx)
+						})
+					})
+				}),
+			)
+		}).
+		NegativeButton("Cancel", func() {
+			pg.autoPurchase.SetChecked(false)
+		}).
+		PositiveButton("Confirm", func(password string, pm *modal.PasswordModal) bool {
+			if !pg.WL.MultiWallet.IsConnectedToDecredNetwork() {
+				pg.Toast.NotifyError(values.String(values.StrNotConnected))
+				pm.SetLoading(false)
+				pg.updateTBToggle()
+				return false
+			}
+
+			go func() {
+				vsp, err := pg.WL.MultiWallet.NewVSPClient(host, walID, uint32(acctNum))
+				if err != nil {
+					pg.Toast.NotifyError(err.Error())
+					pm.SetLoading(false)
+					return
+				}
+
+				err = vsp.StartTicketBuyer(b2m, []byte(password))
+				if err != nil {
+					if err.Error() == dcrlibwallet.ErrInvalidPassphrase {
+						pm.SetError("Invalid password")
+						pm.SetLoading(false)
+					} else {
+						pg.Toast.NotifyError(err.Error())
+					}
+					return
+				}
+				time.AfterFunc(time.Second*3, func() {
+					pg.updateTBToggle()
+				})
```

Comment:
Why after 3 seconds? Maybe add a comment here.
```suggestion
				time.AfterFunc(time.Second*3, pg.updateTBToggle)
```